server/channels/api4/channel.go

```diff
@@ -1366,7 +1366,12 @@ func deleteChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		if *c.App.Config().ServiceSettings.EnableAPIChannelDeletion {
 			err = c.App.PermanentDeleteChannel(c.AppContext, channel)
 		} else {
-			err = model.NewAppError("deleteChannel", "api.user.delete_channel.not_enabled.app_error", nil, "channelId="+c.Params.ChannelId, http.StatusUnauthorized)
+			key := "api.user.delete_channel.not_enabled.app_error"
+			user, usrErr := c.App.GetUser(c.AppContext.Session().UserId)
+			if usrErr == nil && user != nil && user.IsSystemAdmin() {
+				key = "api.user.delete_channel.not_enabled.for_admin.app_error" // More verbose error message for system admins
```

Comment:
Arg, the manual assignment of the translation key is causing issues with our tooling. Can you just use `model.NewAppError` to create the separate errors? Thanks!

Reply:
ok